Add issue templates

ChristianMurphy opened the pull request:
based off https://github.com/remarkjs/.github/tree/c669bbe44cd9ed1960e36c5fb172f4d03f7911c2/.github/ISSUE_TEMPLATE

* replaces references to remark with xdm.
* removes links to github discussions since they aren't enabled on this repo, and instead adds a question template
* Adds potential starts for codesandbox and stackblitz (these appear to be not fully functional due to ESM support gaps in these platforms, alternative ideas are welcome)

wooorm reviewed (commented):
Good idea!

- I’d say don’t add labels: users aren’t always great at that
- Half of unified now has newer templates (e.g., https://github.com/micromark/.github/tree/main/.github/ISSUE_TEMPLATE). Maybe some of that can be used here, except for the bb references (I’ve also thought of adding the new labels + bb to my own repos)

ChristianMurphy commented:
Thanks for the feedback @wooorm!
Updated to use the micromark template, also updated the codesandbox to use evalutate sync which does appear to work consistently in the sandboxed env.

ChristianMurphy commented on .github/ISSUE_TEMPLATE/1-bug.md:
```diff
@@ -0,0 +1,64 @@
+
+### Initial checklist
+
+*   [ ] I read the support docs <!-- https://github.com/micromark/.github/blob/main/support.md -->
+*   [ ] I read the contributing guide <!-- https://github.com/micromark/.github/blob/main/contributing.md -->
+*   [ ] I agree to follow the code of conduct <!-- https://github.com/micromark/.github/blob/main/code-of-conduct.md -->
+*   [ ] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=user%3Amicromark&type=Issues -->
```
these links don't have an xdm equivalent yet, left them pointing to micromark for now.

wooorm replied:
```suggestion
*   [ ] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=repo%3Awooorm%2Fxdm&type=Issues -->
```

wooorm commented on .github/ISSUE_TEMPLATE/2-feature.md:
```diff
@@ -0,0 +1,48 @@
+*   [ ] I read the support docs <!-- https://github.com/micromark/.github/blob/main/support.md -->
+*   [ ] I read the contributing guide <!-- https://github.com/micromark/.github/blob/main/contributing.md -->
+*   [ ] I agree to follow the code of conduct <!-- https://github.com/micromark/.github/blob/main/code-of-conduct.md -->
+*   [ ] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=user%3Amicromark&type=Issues -->
```
```suggestion
*   [ ] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=repo%3Awooorm%2Fxdm&type=Issues -->
```

wooorm reviewed (commented):
Nice! I think this one link can be updated, but otherwise perfect!

wooorm commented:
Awesome!
